### ISSUE TITLE
Typescript definitions?

### ISSUE THREAD

**Issue by pulverize** [FortAwesome/react-fontawesome#17]
I'm using TypeScript for typing in my React app, but I don't have the level of familiarity with TS to write the defs myself.

**Comment by mlwilkerson** [comment]
@pulverize I've been working on getting TypeScript definitions added to the core fontawesome API, and then a thin layer in the React (and Angular) components that sits upon the underlying types. Coming soon.

**Comment by mlwilkerson** [comment]
@pulverize, see PR #46. It's on the way.

**Comment by szmarci** [comment]
@mlwilkerson do you have any news on when your PR gonna make it into a release?

**Comment by mlwilkerson** [comment]
I just released @fortawesome/fontawesome 1.1.1 (the base library) yesterday with TypeScript support. I'm shaking out the bugs in it now in [this discussion](https://github.com/FortAwesome/Font-Awesome-Pro/pull/964) and now [this one](https://github.com/FortAwesome/Font-Awesome-Pro/issues/977).

I think that work will be done today, and then the next priority is to release this React component with the type definitions, which are a thin layer on top of those in the base library.

So, I expect it to be done today or tomorrow.

**Comment by sheam** [comment]
@mlwilkerson Any chance this comes out in the next week?

**Comment by mlwilkerson** [comment]
Yes, really close. @robmadole and I just concluded a phone call which culminated in clarifying action steps for delivering the behind-the-scenes work as pre-releases, so that those of you who have been waiting can get some forward momentum.

Needless to say, the TypeScript work has driven more underlying changes than we'd expected. All for the better in the long run we're sure, but more than we expected when we first embarked on the journey.

**Comment by sheam** [comment]
Thanks for the update it is appreciated.

**Comment by kirkbross** [comment]
Are type definitions included or is there an @types/react-fontawesome package? I am getting _'JSX element type FontAwesomeIcon does not have any construct or call signatures.'_
